packages/arb-bridge-eth/contracts/rollup/Node.sol

```diff
@@ -21,57 +21,63 @@ pragma experimental ABIEncoderV2;
 
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
-struct NodeProps {
+struct Node {
     // Hash of the state of the chain as of this node
     bytes32 stateHash;
-
     // Hash of the data that can be challenged
     bytes32 challengeHash;
-
     // Hash of the data that will be committed if this node is confirmed
     bytes32 confirmData;
-
     // Index of the node previous to this one
-    uint256 prev;
-
+    uint256 prevNum;
     // Deadline at which this node can be confirmed
     uint256 deadlineBlock;
-
     // Deadline at which a child of this node can be confirmed
     uint256 noChildConfirmedBeforeBlock;
-
     // Number of stakers staked on this node. This includes real stakers and zombies
     uint256 stakerCount;
-
-    // Address of the rollup contract to which this node belongs
-    address rollup;
-
     // This value starts at zero and is set to a value when the first child is created. After that it is constant until the node is destroyed or the owner destroys pending nodes
     uint256 firstChildBlock;
-
     // The number of the latest child of this node to be created
     uint256 latestChildNumber;
 }
 
-struct Node {
-    // Mapping of the stakers staked on this node with true if they are staked. This includes real stakers and zombies
-    mapping(address => bool) stakers;
-    
-    // All other Node data
-    NodeProps props;
-}
-
 /**
- * @notice Utility functions for NodeProps
+ * @notice Utility functions for Node
  */
-library NodePropsLib {
+library NodeLib {
     using SafeMath for uint256;
 
+    /**
+     * @notice Initialise a Node
+     * @param _stateHash Initial value of stateHash
+     * @param _challengeHash Initial value of challengeHash
+     * @param _confirmData Initial value of confirmData
+     * @param _prevNum Initial value of prevNum
+     * @param _deadlineBlock Initial value of deadlineBlock
+     */
+    function initialize(
```

Comment:
If we wanted to initialise without copying to memory, we could probably pass in a storage pointer of an empty struct then write to it, ie
```solidity
function initialize(
    Node storage self,
    ...
)
```

There also is another notation for creating structs which I particularly like, but it prob has the same gas cost
```solidity
Node memory node = Node({
    stateHash: _stateHash,
    ...
});
```

Reply:
Yeah, I think we'll only save a couple 100 gas doing that. We also may need an extra local var in one of the calling functions, and I'm not sure we have the stack space. Maybe lets leave this for a later improvement?

Regarding the second notation, I prefer it too, however it wouldn't let me use it without specifying all the parameter and there's a bunch we dont actually need to initialise.